### PR TITLE
[OAP-1804][Doc]Reorg docs for SQL Data Source Cache

### PR DESCRIPTION
**Pull request by HongW2019**
## What changes were proposed in this pull request?
As to OAP SQL Index and Data Source Cache, we use external cache strategy instead of vmemcache strategy as main introduction line in User-Guide.md from OAP 1.0.0, move the vmemcache introduction to Advanced-Configuration.md. 
Then this change leads to modificaitons on PMem config both in User-Guide.md and Advanced-Configuration.md. 

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by zhixingheyi-tian** [comment]
@HongW2019  Please add description in Conversation

**Comment by zhixingheyi-tian** [comment]
@jikunshang @yma11 

Hong now is adjusting Plasma section Guide, please have  a review.

Thanks,

File: oap-cache/oap/docs/Advanced-Configuration.md [hunk 1]
```diff
@@ -75,186 +125,134 @@ spark.sql.oap.cache.memory.manager           kmem
...
-To use this strategy, follow [prerequisites](./User-Guide.md#prerequisites-1) to set up PMem hardware. 
-
-Besides, with BIOS configuration settings below, PMem could get noticeable performance gain, especially on cross socket write path.
-
-```
-Socket Configuration -> Memory Configuration -> NGN Configuration -> Snoopy mode for AD : enabled
-Socket configuration -> Intel UPI General configuration -> Stale Atos :  Disabled
-``` 
 
+# Enable OAP jar in Spark
```
Comment (jikunshang) [inline oap-cache/oap/docs/Advanced-Configuration.md:141]: Enable OAP extension in Spark?

File: oap-cache/oap/docs/Advanced-Configuration.md [hunk 2]
```diff
@@ -75,186 +125,134 @@ spark.sql.oap.cache.memory.manager           kmem
...
-# relative path of the jar
-spark.executor.extraClassPath     ./oap-cache-<version>-with-spark-<version>.jar:./oap-common-<version>-with-spark-<version>.jar:./arrow-plasma-0.17.0.jar
-# absolute path of the jar on your working node
-spark.driver.extraClassPath       $HOME/miniconda2/envs/oapenv/oap_jars/oap-cache-<version>-with-spark-<version>.jar:$HOME/miniconda2/envs/oapenv/oap_jars/oap-common-<version>-with-spark-<version>.jar:$HOME/miniconda2/envs/oapenv/oap_jars/arrow-plasma-0.17.0.jar
+- Make sure [Vmemcache](https://github.com/pmem/vmemcache) library has been installed on every cluster worker node if vmemcache strategy is chosen for PMem cache. If you have finished [OAP-Installation-Guide](../../docs/OAP-Installation-Guide.md), vmemcache library will be automatically installed by Conda.
+  
+  Or you can follow the [build/install](./Developer-Guide.md#build-and-install-vmemcache) steps and make sure `libvmemcache.so` exist in `/lib64` directory in each worker node.
```
Comment (jikunshang) [inline oap-cache/oap/docs/Advanced-Configuration.md:157]: on each worker node?

**Review by jikunshang — APPROVED** [review]
LGTM. A few minor comments left.

File: oap-cache/oap/docs/Advanced-Configuration.md [hunk 3]
```diff
@@ -75,186 +125,134 @@ spark.sql.oap.cache.memory.manager           kmem
...
-# enable numa
-spark.yarn.numa.enabled                                  true
-spark.executorEnv.MEMKIND_ARENA_NUM_PER_KIND             1
-spark.sql.oap.orc.binary.cache.enabled                   true 
-spark.oap.cache.strategy                                 noevict 
+spark.yarn.numa.enabled                                      true
+spark.executorEnv.MEMKIND_ARENA_NUM_PER_KIND                 1
+# for Parquet file format, enable binary cache
+spark.sql.oap.parquet.binary.cache.enabled                   true 
+for ORC file format, enable binary cache
```
[the remark this answers is not in the document]
Comment (HongW2019) [reply oap-cache/oap/docs/Advanced-Configuration.md]: have modified, thx.

**Pull request merged**